server/src/main/java/org/candlepin/async/tasks/RefreshPoolsForProductJob.java

```diff
@@ -0,0 +1,150 @@
+import org.candlepin.controller.PoolManager;
+import org.candlepin.controller.Refresher;
+import org.candlepin.model.Product;
+import org.candlepin.model.ProductCurator;
+import org.candlepin.pinsetter.core.model.JobStatus;
+import org.candlepin.service.OwnerServiceAdapter;
+import org.candlepin.service.SubscriptionServiceAdapter;
+
+import com.google.inject.Inject;
+
+import java.util.Objects;
+
+
+/**
+ * RefreshPoolsForProductJob
+ */
+public class RefreshPoolsForProductJob implements AsyncJob {
+
+    private final PoolManager poolManager;
+    private final ProductCurator productCurator;
+    private final SubscriptionServiceAdapter subAdapter;
+    private final OwnerServiceAdapter ownerAdapter;
+
+    public static final String JOB_KEY = "REFRESH_POOLS_FOR_PRODUCT";
+    private static final String JOB_NAME = "refresh_pools_for_product";
+    private static final String LAZY_REGEN = "lazy_regen";
+
+    @Inject
+    public RefreshPoolsForProductJob(
+        final ProductCurator productCurator,
+        final PoolManager poolManager,
+        final SubscriptionServiceAdapter subAdapter,
+        final OwnerServiceAdapter ownerAdapter) {
+        this.poolManager = Objects.requireNonNull(poolManager);
+        this.productCurator = Objects.requireNonNull(productCurator);
+        this.subAdapter = Objects.requireNonNull(subAdapter);
+        this.ownerAdapter = Objects.requireNonNull(ownerAdapter);
+    }
+
+    @Override
+    public String execute(final JobExecutionContext context) {
+        final JobArguments args = context.getJobArguments();
+
+        final String productUuid = args.getAsString(JobStatus.TARGET_ID);
+        final Boolean lazy = args.getAsBoolean(LAZY_REGEN);
+        final StringBuilder result = new StringBuilder();
+
+        final Product product = this.productCurator.get(productUuid);
+
+        if (product != null) {
+            final Refresher refresher = poolManager.getRefresher(this.subAdapter, this.ownerAdapter, lazy);
+
+            refresher.add(product);
+            refresher.run();
+
+            result.append("Pools refreshed for product: ")
+                .append(productUuid)
+                .append("\n");
+        }
+        else {
+            result.append("Unable to refresh pools for product \"")
+                .append(productUuid)
+                .append("\": Could not find a product with the specified UUID");
+        }
+
+        return result.toString();
+    }
+
+    /**
+     * Creates a JobConfig configured to execute the export job. Callers may further manipulate
+     * the JobConfig as necessary before queuing it.
+     *
+     * @return
+     *  a JobConfig instance configured to execute the export job
+     */
+    public static RefreshPoolsForProductConfig createJobConfig() {
+        return new RefreshPoolsForProductConfig();
+    }
+
+    /**
+     * Job configuration object for the export job
+     */
+    public static class RefreshPoolsForProductConfig extends JobConfig {
```

Comment:
Yeah it is very long, but it is also tautological. Since RefreshPoolsForProductJobConfig is nested class we have to use name of its outer class. E.g.

> RefreshPoolsForProductJob.RefreshPoolsForProductJobConfig config

Since the job name is already there and has to be present we could remove it from the name of JobConfig. E.g.

> RefreshPoolsForProductJob.Config config

What do you think?